packages/itwinui-react/src/core/Tooltip/Tooltip.tsx

```diff
@@ -16,46 +54,141 @@ export type TooltipProps = {
    * Element to have tooltip on. Has to be a valid JSX element and needs to forward its ref.
    * If not specified, the `reference` prop should be used instead.
    */
-  children?: JSX.Element;
-} & Omit<PopoverProps, 'className'> &
-  Omit<CommonProps, 'title'>;
+  children?: React.ReactNode;
+  /**
+   * Element to portal tooltip to.
+   * Portals to ThemeProvider portalContainerRef by default.
+   */
+  portal?: { to: HTMLElement };
+};
 
-/**
- * Basic tooltip component to display informative content when an element is hovered or focused.
- * Uses the {@link Popover} component, which is a wrapper around [tippy.js](https://atomiks.github.io/tippyjs).
- * @example
- * <Tooltip content='tooltip text' placement='top'><div>Hover here</div></Tooltip>
- * @example
- * const buttonRef = React.useRef();
- * ...
- * <Button ref={buttonRef} />
- * <Tooltip content='tooltip text' reference={buttonRef} />
- */
-export const Tooltip = (props: TooltipProps) => {
-  const { content, children, className, style, visible, ref, id, ...rest } =
-    props;
+const useTooltip = ({
+  placement = 'top',
+  visible: controlledOpen,
+  toggleVisible: setControlledOpen,
+  followTrigger = false,
+}: TooltipOptions = {}) => {
+  const [uncontrolledOpen, setUncontrolledOpen] = React.useState(false);
+
+  const open = controlledOpen ?? uncontrolledOpen;
+  const setOpen = setControlledOpen ?? setUncontrolledOpen;
+
+  const data = useFloating({
+    placement,
+    open,
+    onOpenChange: setOpen,
+    whileElementsMounted: (referenceEl, floatingEl, update) =>
+      autoUpdate(referenceEl, floatingEl, update, {
+        animationFrame: followTrigger,
+      }),
+    middleware: [offset(5), flip(), shift()],
+  });
+
+  const context = data.context;
+
+  const hover = useHover(context);
+  const focus = useFocus(context, {
+    enabled: controlledOpen == null,
+  });
+  const click = useClick(context);
+
+  const dismiss = useDismiss(context);
+  const role = useRole(context, { role: 'tooltip' });
+
+  const interactions = useInteractions([hover, focus, click, dismiss, role]);
 
+  return React.useMemo(
+    () => ({
+      open,
+      setOpen,
+      ...interactions,
+      ...data,
+    }),
+    [open, setOpen, interactions, data],
+  );
+};
+
+const TooltipContext = React.createContext<
```

Comment:
if we are not exposing subcomponents, then do we need context? would be simpler to do it all in one component.

Reply:
For future, I think we want to have subcomponents for Tooltip so it would be easy transition from props to that.
Keeping props for now for easier transition.
*** 
Though I don't know, users would need to update props anyways so not sure of the benefit to keep old API (which is only content and children props).

Reply:
the benefit of `children` and `content` is convenience.

It is easier to do this:
```jsx
<Tooltip content='close'>
 <Button>❌</Button>
</Tooltip>
```
than this:
```jsx
<Tooltip>
 <Tooltip.Trigger>
  <Button>❌</Button>
 </Tooltip.Trigger>
 <Tooltip.Content>
  close
 </Tooltip.Content>
</Tooltip>
```

whereas I'm not sure about the benefit of subcomponents here because the trigger does not render anything, it does a cloneElement, so there is only one "real" element.

also, right now we are using `children` as trigger, so it locks us out of adding subcomponents inside `<Tooltip>`. which means if we actually want subcomponents, we might need to add a `<Tooltip.Wrapper>`. see https://github.com/iTwin/iTwinUI/pull/1328#issuecomment-1582974467